packages/lib-classifier/src/plugins/drawingTools/experimental/models/marks/FreehandLine/FreehandLine.js

```diff
@@ -3,277 +3,646 @@ import { FreehandLine as FreehandLineComponent } from '@plugins/drawingTools/com
 import { Mark } from '@plugins/drawingTools/models/marks'
 import { FreehandLineTool } from '@plugins/drawingTools/models/tools'
 import { FixedNumber } from '@plugins/drawingTools/types/'
-
-const LINE_RESOLUTION = 0.5
-const MINIMUM_POINTS = 20
+import { toJS } from 'mobx'
 
 const SingleCoord = types.model('SingleCoord', {
   x: FixedNumber,
   y: FixedNumber
 })
 
+const ActionType = types.model('ActionType', {
+  type: types.string,
```

Comment:
```suggestion
  type: types.enumeration([
    'append',
    'end',
    'start',
    'splice-append',
    'splice-close-point',
    'splice-drag-point',
    'splice-end'
  ]),
```

I think best to specify the allowed types, rather than accept any arbitrary string.